### ISSUE TITLE
Realm is not defined in debug modus

### ISSUE THREAD

**Issue by Acetyld** [realm/realm-js#3991]
### How frequently does the bug occur?

All the time

### Description

Using any kind of debugger is causing:
LogBox.js:173 Possible Unhandled Promise Rejection (id: 0):
ReferenceError: Realm is not defined
ReferenceError: Realm is not defined

Chrome, 

### Stacktrace & log output

```shell
index.bundle?platfor…g.reactjs.na…:23154 Possible Unhandled Promise Rejection (id: 0):
ReferenceError: Realm is not defined
ReferenceError: Realm is not defined
    at Proxy.method (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.Realm:120435:90)
    at Function.from (<anonymous>)
    at _iterableToArray (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.Realm:3511:116)
    at _toConsumableArray (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.Realm:3495:131)
    at http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.Realm:109584:50
    at tryCallOne (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.Realm:28713:14)
    at http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.Realm:28814:17
    at http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.Realm:29405:21
    at _callTimer (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.Realm:29301:9)
    at _callReactNativeMicrotasksPass (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.Realm:29340:7)
```

### Can you reproduce the bug?

Yes, always

### Reproduction Steps

   "react": "17.0.2",
    "react-native": "0.66.0",
    "realm": "^10.8.0"

Any debug modes is causing this, even a fresh project

### Version

    "realm": "^10.8.0",

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

Newest

### Build environment

Which debugger for React Native: ..

**Comment by Acetyld** [comment]
Found a fix using some older github issues
    **I changed:**
    const isSet = (this.constructor && this.constructor.name === "Set") || this instanceof Realm.Set;
    **to:**
    const isSet = (this.constructor && this.constructor.name && this.constructor.name === 'Set');
    
    Inside relam/lib/collection-methods.js.
    
    Can you guys accept the old pull request or fix this ASAP?
    Otherwise no debugger will work.

**Comment by fronck** [comment]
@Acetyld Which debugger are you using to trigger this issue?

**Comment by Acetyld** [comment]
It happens with every debugger. 
Safara, chrome  debugger
also with chrome debugger and even with flipper.,

**Comment by takameyer** [comment]
@Acetyld Debugging in Safari should work if you do not activate debugging mode (do not press `Debug with Chrome`).  Theres a [good blog post](http://blog.nparashuram.com/2019/10/debugging-react-native-ios-apps-with.html) on how to do this.  Let us know if works.

**Comment by Acetyld** [comment]
Yhea i tried that. but its not due to chrome debugger. Its als in flipper
**I changed:**
const isSet = (this.constructor && this.constructor.name === "Set") || this instanceof Realm.Set;
**to:**
const isSet = (this.constructor && this.constructor.name && this.constructor.name === 'Set');

This fixed the issue.

I also saw a other post complaning about the pull you deleted this https://github.com/realm/realm-js/pull/3843.

**Comment by fronck** [comment]
@Acetyld We haven't been able to reproduce the issue with either lldb or Safari.  We do have a code cleanup PR in #4109 .
Would you be able to provide reproduction code, or try out the fix in the PR to make sure it fixes the issue?

**Comment by fronck** [comment]
This issue should be fixed by #4109 .
Thanks for reporting.
